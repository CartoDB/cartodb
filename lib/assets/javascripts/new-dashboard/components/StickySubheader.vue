<template>
  <section class="sticky-subheader" :class="{ 'is-visible': $props.isVisible }">
    <div class="container subheader-container">
      <slot />
    </div>
  </section>
</template>

<script>
export default {
  name: 'StickySubheader',
  props: {
    isVisible: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="scss" scoped>
@import 'stylesheets/new-dashboard/variables';

.sticky-subheader {
  display: flex;
  position: fixed;
  z-index: 2;
  top: 0;
  left: 0;
  justify-content: center;
  width: 100%;
  height: 64px;
  transition: transform 200ms cubic-bezier(0.4, 0.01, 0.165, 0.99);
  border-bottom: 1px solid $grey;
  background-color: $white;

  &.is-visible {
    transform: translate3d(0, 64px, 0);
  }
}

.subheader-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 0 20px;
}
</style>
