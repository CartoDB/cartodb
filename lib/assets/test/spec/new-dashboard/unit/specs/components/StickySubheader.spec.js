import { shallowMount } from '@vue/test-utils';
import StickySubheader from 'new-dashboard/components/StickySubheader';

describe('StickySubheader.vue', () => {
  it('should render correct contents', () => {
    const stickySubheader = shallowMount(StickySubheader, {
      propsData: { isVisible: false },
      slots: {
        default: '<div class="default-slot"></div>'
      }
    });
    expect(stickySubheader).toMatchSnapshot();
  });

  it('should have is-visible class when isVisible is true', () => {
    const stickySubheader = shallowMount(StickySubheader, { propsData: { isVisible: true } });
    expect(stickySubheader).toMatchSnapshot();
  });
});
