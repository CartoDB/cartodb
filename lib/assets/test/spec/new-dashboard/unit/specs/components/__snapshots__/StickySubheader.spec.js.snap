// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickySubheader.vue should have is-visible class when isVisible is true 1`] = `
<section class="sticky-subheader is-visible">
  <div class="container subheader-container"></div>
</section>
`;

exports[`StickySubheader.vue should render correct contents 1`] = `
<section class="sticky-subheader">
  <div class="container subheader-container">
    <div class="default-slot"></div>
  </div>
</section>
`;
